app/index.tsx

```diff
@@ -3,13 +3,14 @@ import { Redirect } from 'expo-router';
 import React from 'react';
 
 function RootScreen() {
-    const { isLoggedIn } = useSession();
+    console.log('RootScreen');
+    const { isLoggedIn, initialized } = useSession();
 
-    if (!isLoggedIn) {
-        return <Redirect href="/signup" />;
+    if (!initialized) {
```

Comment:
return a loading component instead of redirecting, this can cause have bad performance footprint